src/routeDefinitions/useLocalizedRoutes.tsx

```diff
@@ -0,0 +1,297 @@
+import { LangCode, supportedLanguageCodes } from '@/i18n/config';
+import { AuthGuard } from '@/routes/AuthGuard';
+import {
+  AccessibilityStatement,
+  BasicInformation,
+  CookiePolicy,
+  DataSources,
+  PrivacyPolicy,
+  TermsOfService,
+} from '@/routes/BasicInformation';
+import { Home } from '@/routes/Home';
+import {
+  JobOpportunity,
+  Competences as JobOpportunityCompetences,
+  Overview,
+  loader as jobOpportunityLoader,
+} from '@/routes/JobOpportunity';
+import { Favorites, Preferences, Profile, SomethingElse } from '@/routes/Profile';
+import { Competences, competencesLoader } from '@/routes/Profile/Competences';
+import { EducationHistoryV2, loader as educationHistoryLoader } from '@/routes/Profile/EducationHistory';
+import { FreeTimeActivities, loader as freeTimeActivitiesLoader } from '@/routes/Profile/FreeTimeActivities';
+import { WorkHistory, loader as workHistoryLoader } from '@/routes/Profile/WorkHistory';
+import { ErrorElement, Root, loader as rootLoader } from '@/routes/Root';
+import { Goals, Instructions, Interests, Restrictions, Tool, Competences as ToolCompetences } from '@/routes/Tool';
+import {
+  HowDoIGiveFeedback,
+  HowDoIUseTheService,
+  UserGuide,
+  WhatIsTheService,
+  WhereCanIGetMoreHelp,
+  WhoIsTheServiceFor,
+  WhoProvidesTheService,
+} from '@/routes/UserGuide';
+import { useTranslation } from 'react-i18next';
+import { RouteObject, generatePath, matchRoutes, redirect } from 'react-router-dom';
+
+const useLocalizedRoutes = () => {
+  const { i18n } = useTranslation();
+  const jobOpportunityRouteId = (lang: LangCode) => `job-opportunity-${lang}`;
+
+  const flattenRoutes = (routes: RouteObject[], parentPath = ''): RouteObject[] => {
+    return routes.reduce((flatRoutes: RouteObject[], route: RouteObject) => {
+      // eslint-disable-next-line sonarjs/no-nested-template-literals
+      const fullPath = `${parentPath}${route.path ? `/${route.path}` : ''}`.replace(/\/+/g, '/');
+
+      const { children, ...routeWithoutChildren } = route;
+
+      flatRoutes.push({
+        ...routeWithoutChildren,
+        path: fullPath,
+      });
+
+      if (children) {
+        flatRoutes.push(...flattenRoutes(children, fullPath));
+      }
+
+      return flatRoutes;
+    }, []);
+  };
+
+  const extractPaths = (routes: RouteObject[], basePath = '') => {
+    let paths: string[] = [];
+
+    routes.forEach((route) => {
+      if (route.path) {
+        const fullPath = `${basePath}${basePath === '' ? '' : '/'}${route.path}`;
+
+        paths.push(fullPath);
+
+        // If the route has children, recursively add their paths
+        if (route.children) {
+          paths = [...paths, ...extractPaths(route.children, fullPath)];
+        }
+      }
+    });
+
+    return paths;
+  };
```

Comment:
I'm not sure what happens in these functions. Could there possibly be a unit test?

Reply:
Splitted the code, added documentation and wrote tests.